src/core/thread/mesh_forwarder.cpp

```diff
@@ -251,18 +253,31 @@ void MeshForwarder::ScheduleTransmissionTask()
             continue;
         }
 
-        mSendMessage = child.mIndirectSendMessage;
+        mSendMessage = child.mIndirectSendInfo.mMessage;
 
         if (mSendMessage == NULL)
         {
             mSendMessage = GetIndirectTransmission(child);
-            child.mIndirectSendMessage = mSendMessage;
-            child.mFragmentOffset = 0;
+
+            child.mIndirectSendInfo.mMessage = mSendMessage;
+            child.mIndirectSendInfo.mFragmentOffset = 0;
+            child.mIndirectSendInfo.mTxAttemptCounter = 0;
```

Comment:
I was considering this... but at the end I felt it would be good to update all the `mIndirectSendInfo` fields from one place...